cirq-google/cirq_google/serialization/circuit_serializer.py

```diff
@@ -416,7 +443,15 @@ def _deserialize_circuit(
         deserialized_constants: List[Any],
     ) -> cirq.Circuit:
         moments = []
+        constant_moments: dict[int, cirq.Moment] = {}
         for moment_proto in circuit_proto.moments:
+            if constant_index := moment_proto.moment_constant_index:
```

Comment:
should we split this into creation and condition? we use the `constant_index` object outside the if condition.. I couldn't find anything explaining when object references defined with `:=` become invalid ... if it's anything like C++ their scope would be only within the `if` condition

Reply:
No, the walrus operator just does assignment and condition check.  It doesn't have a scope constrained to only the if statement.